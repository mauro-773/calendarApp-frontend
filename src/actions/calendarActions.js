import Swal from 'sweetalert2';
import { fetchWithToken } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';
import { TYPES } from '../types/types';

export const addEventAction = (event) => async (dispatch, getState) => {
   const { uid, name } = getState().auth;
   try {
      const res = await fetchWithToken('events', event, 'POST');
      const body = await res.json();

      if (body.ok) {
         event.id = body.event.id;
         event.user = {
            _id: uid,
            name,
         };
         dispatch(addEvent(event));
      }
   } catch (error) {
      console.log(error);
   }
};

const addEvent = (event) => ({
   type: TYPES.ADD_NEW_EVENT,
   payload: event,
});

export const loadEventsAction = () => async (dispatch) => {
   try {
      const res = await fetchWithToken('events');
      const body = await res.json();
      const events = prepareEvents(body.events);
      dispatch(eventLoaded(events));
   } catch (error) {
      console.log(error);
   }
};

const eventLoaded = (events) => ({
   type: TYPES.EVENT_LOADED,
   payload: events,
});

export const setActiveAction = (event) => ({
   type: TYPES.SET_ACTIVE_EVENT,
   payload: event,
});

export const clearActiveAction = () => ({
   type: TYPES.CLEAR_ACTIVE_EVENT,
});

export const updateEventAction = (event) => async (dispatch) => {
   try {
      const res = await fetchWithToken(`events/${event.id}`, event, 'PUT');
      const body = await res.json();

      body.ok
         ? dispatch(updateEvent(event))
         : Swal.fire('Error', body.msg, 'error');
   } catch (error) {
      console.log(error);
   }
};

const updateEvent = (event) => ({
   type: TYPES.UPDATE_EVENT,
   payload: event,
});

export const deleteEventAction = () => async (dispatch, getState) => {
   const { id } = getState().calendar.activeEvent;
   try {
      const res = await fetchWithToken(`events/${id}`, {}, 'DELETE');
      const body = await res.json();

      body.ok ? dispatch(deleteEvent()) : Swal.fire('Error', body.msg, 'error');
   } catch (error) {
      console.log(error);
   }
};

const deleteEvent = () => ({
   type: TYPES.DELETE_EVENT,
});

export const clearEventsAction = () => ({
   type: TYPES.EVENT_LOGOUT,
});
