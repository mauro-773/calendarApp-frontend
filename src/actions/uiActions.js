import { TYPES } from '../types/types';

export const openModalAction = () => ({
   type: TYPES.OPEN_MODAL,
   payload: true,
});

export const closeModalAction = () => ({
   type: TYPES.CLOSE_MODAL,
   payload: false,
});
