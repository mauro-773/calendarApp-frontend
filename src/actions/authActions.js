import { fetchWithoutToken, fetchWithToken } from '../helpers/fetch';
import { TYPES } from '../types/types';
import Swal from 'sweetalert2';
import { clearEventsAction } from './calendarActions';

const saveTokenLS = (token) => {
   localStorage.setItem('token', token);
   localStorage.setItem('token-init-date', new Date().getTime());
};

const showMSGError = (msg) => {
   Swal.fire('Error', msg, 'error');
};

export const startLoginAction = (email, password) => async (dispatch) => {
   try {
      const res = await fetchWithoutToken('auth', { email, password }, 'POST');
      const body = await res.json();

      if (body.ok) {
         saveTokenLS(body.token);
         dispatch(login({ uid: body.uid, name: body.name }));
      } else {
         showMSGError(body.msg);
      }
   } catch (error) {
      console.log(error);
   }
};

const login = (user) => ({
   type: TYPES.START_LOGIN,
   payload: user,
});

export const startRegisterAction = (name, email, password) => async (
   dispatch
) => {
   const newUser = { name, email, password };
   try {
      const res = await fetchWithoutToken('auth/new', newUser, 'POST');
      const body = await res.json();

      if (body.ok) {
         saveTokenLS(body.token);
         dispatch(login({ uid: body.uid, name: body.name }));
      } else {
         showMSGError(body.msg);
      }
   } catch (error) {
      console.log(error);
   }
};

export const startCheckingAction = () => async (dispatch) => {
   try {
      const res = await fetchWithToken('auth/renew');
      const body = await res.json();

      if (body.ok) {
         saveTokenLS(body.token);
         dispatch(login({ uid: body.uid, name: body.name }));
      } else {
         dispatch(checkingFinish());
      }
   } catch (error) {
      console.log(error);
   }
};

const checkingFinish = () => ({ type: TYPES.AUTH_CHECKING_FINISH });

export const startLogoutAction = () => (dispatch) => {
   localStorage.clear();
   dispatch(logout());
   dispatch(clearEventsAction());
};

const logout = () => ({ type: TYPES.LOGOUT });
