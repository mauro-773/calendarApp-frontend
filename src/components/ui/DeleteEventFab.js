import { useDispatch } from 'react-redux';
import { deleteEventAction } from '../../actions/calendarActions';

const DeleteEventFab = () => {
   const dispatch = useDispatch();

   const handleDeleteEvent = () => {
      dispatch(deleteEventAction());
   };

   return (
      <button className="btn btn-danger fab-danger" onClick={handleDeleteEvent}>
         <i className="fas fa-trash"></i>
         <span> Borrar Evento</span>
      </button>
   );
};

export default DeleteEventFab;
