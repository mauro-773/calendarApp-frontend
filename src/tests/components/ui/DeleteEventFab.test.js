import DeleteEventFab from '../../../components/ui/DeleteEventFab';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { deleteEventAction } from '../../../actions/calendarActions';

import '@testing-library/jest-dom';

jest.mock('../../../actions/calendarActions', () => ({
   deleteEventAction: jest.fn(),
}));

// Configuraci√≥n del store de redux
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
const store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
   <Provider store={store}>
      <DeleteEventFab />
   </Provider>
);

describe('Test en <DeleteEventFab />', () => {
   test('Debe hacer el match con el snapshot', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Debe llamar a deleteEventAction al hacer click', () => {
      wrapper.find('button').prop('onClick')();
      expect(deleteEventAction).toHaveBeenCalled();
   });
});
