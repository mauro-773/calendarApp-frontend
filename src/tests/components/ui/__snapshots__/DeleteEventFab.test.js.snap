// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <DeleteEventFab /> Debe hacer el match con el snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar Evento
  </span>
</button>
`;
